error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJDurationRules.java

```diff
@@ -0,0 +1,119 @@
+
+/**
+ * Refaster rules related to AssertJ assertions over {@link Duration}s.
+ *
+ * <p>These rules simplify and improve the readability of Duration assertions by using the more
+ * specific AssertJ Duration assertion methods instead of generic assertions on the Duration's
+ * numeric values.
+ */
+@OnlineDocumentation
+final class AssertJDurationRules {
+  private AssertJDurationRules() {}
+
+  static final class AbstractDurationAssertHasNanos {
+    @BeforeTemplate
+    AbstractLongAssert<?> before(Duration duration, long nanos) {
+      return assertThat(duration.toNanos()).isEqualTo(nanos);
+    }
+
+    @AfterTemplate
+    AbstractDurationAssert<?> after(Duration duration, long nanos) {
+      return assertThat(duration).hasNanos(nanos);
+    }
+  }
+
+  static final class AbstractDurationAssertHasMillis {
+    @BeforeTemplate
+    AbstractLongAssert<?> before(Duration duration, long millis) {
+      return assertThat(duration.toMillis()).isEqualTo(millis);
+    }
+
+    @AfterTemplate
+    AbstractDurationAssert<?> after(Duration duration, long millis) {
+      return assertThat(duration).hasMillis(millis);
+    }
+  }
+
+  static final class AbstractDurationAssertHasSeconds {
+    @BeforeTemplate
+    AbstractLongAssert<?> before(Duration duration, long seconds) {
+      return assertThat(duration.toSeconds()).isEqualTo(seconds);
+    }
+
+    @AfterTemplate
+    AbstractDurationAssert<?> after(Duration duration, long seconds) {
+      return assertThat(duration).hasSeconds(seconds);
+    }
+  }
+
+  static final class AbstractDurationAssertHasMinutes {
+    @BeforeTemplate
+    AbstractLongAssert<?> before(Duration duration, long minutes) {
+      return assertThat(duration.toMinutes()).isEqualTo(minutes);
+    }
+
+    @AfterTemplate
+    AbstractDurationAssert<?> after(Duration duration, long minutes) {
+      return assertThat(duration).hasMinutes(minutes);
+    }
+  }
+
+  static final class AbstractDurationAssertHasHours {
+    @BeforeTemplate
+    AbstractLongAssert<?> before(Duration duration, long hours) {
+      return assertThat(duration.toHours()).isEqualTo(hours);
+    }
+
+    @AfterTemplate
+    AbstractDurationAssert<?> after(Duration duration, long hours) {
+      return assertThat(duration).hasHours(hours);
+    }
+  }
+
+  static final class AbstractDurationAssertHasDays {
+    @BeforeTemplate
+    AbstractLongAssert<?> before(Duration duration, long days) {
+      return assertThat(duration.toDays()).isEqualTo(days);
+    }
+
+    @AfterTemplate
+    AbstractDurationAssert<?> after(Duration duration, long days) {
+      return assertThat(duration).hasDays(days);
+    }
+  }
+
+  static final class AbstractDurationAssertIsZero {
+    @BeforeTemplate
+    AbstractBooleanAssert<?> before(Duration duration) {
+      return assertThat(duration.isZero()).isTrue();
+    }
+
+    @AfterTemplate
+    AbstractDurationAssert<?> after(Duration duration) {
+      return assertThat(duration).isZero();
+    }
+  }
```

Comment:
For numeric types we migrate away from `isZero()` in favour of `isEqualTo`([example](https://github.com/PicnicSupermarket/error-prone-support/blob/f5c2943d61250655f00874469263402b5f5cb738/error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJIntegerRules.java#L42-L52)), but for this non-numeric type I suppose the opposite direction does make more sense. Let's then also add a match for `isEqualTo(Duration.ZERO)`.

(I _could_ see us change our mind on this, but then at least we'll remember to update both cases.)

Reply:
Yeah I could also see us go the other way for using the `isZero` specific operators. I know that on the OpenRewrite side, they are preferring this for numeric types.